tiledb/sm/array/array.cc

```diff
@@ -582,6 +582,32 @@ void Array::delete_fragments_list(
   }
 }
 
+shared_ptr<Enumeration> Array::get_enumeration(const std::string& attr_name) {
+  if (!is_open_) {
+    throw ArrayStatusException("Cannot get enumeration; Array is not open");
+  }
+
+  // Do we need to check all previous schema versions if the attribute has
```

Comment:
The core of my questions around array schema is what the intended behavior is for when we're reading old fragments that had previous versions of the schema. My plan for enumerations was to only allow extensions to the list of enumerated values so that reading old fragments is coherent. This particular case is if we have dropped the attribute in the current schema but it exists in a previous version, should we reflect that? My thought was that maybe we go back to the most recent schema with the attribute if necessary but I'm not 100% on whether that's the intended behavior or not.

Reply:
After having worked with this code more. We'll have to change ArraySchemaEvolution to enforce any sort of long term attribute awareness so I'm calling this a problem for another day.